test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -47,16 +47,298 @@ class TestDupAccLogForResv(TestFunctional):
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=8)
+        # Repeating step 5 and 6
+        self.server.restart()
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+    def test_standing_reservation(self):
+        """
+        Test for duplicate records in accounting log for standing reservations
+        on restart of server
+        """
+        # PBS_TZID environment variable must be set, there is no way to set
+        # it through the API call, use CLI instead for this test
+
+        _m = self.server.get_op_mode()
+        if _m != PTL_CLI:
+            self.server.set_op_mode(PTL_CLI)
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        a = {'Resource_List.select': '1:ncpus=1',
+             ATTR_resv_rrule: 'FREQ=MINUTELY;COUNT=2',
+             ATTR_resv_timezone: tzone,
+             ATTR_resv_standing: '1',
+             'reserve_start': time.time() + 5,
+             'reserve_end': time.time() + 60, }
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=6)
+        if _m == PTL_API:
+            self.server.set_op_mode(PTL_API)
+
+        # Verify accounting log before server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=60)
+
+        # Verify accounting log for standing reservation starting second time
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 2)
```

Comment:
if there are two logs for the same rid, how do we differentiate if they are for two different instances?